jwala-services/src/main/java/com/cerner/jwala/service/balancermanager/impl/BalancerManagerHtmlParser.java

```diff
@@ -14,34 +14,38 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(BalancerManagerHtmlParser.class);
 
     public String getUrlPath(final String host, final int httpsPort) {
-        return "https://" + host + ":" +  httpsPort + "/balancer-manager";
+        return "https://" + host + ":" + httpsPort + "/balancer-manager";
     }
 
     public String getWorkerUrlPath(final String host, final int httpsPort, final String balancerName, final String nonce, final String workerName) {
-        return "https://" + host + ":" +  httpsPort + "/balancer-manager" + "?b=" + balancerName + "&nonce=" + nonce + "&w=" + workerName;
+        return "https://" + host + ":" + httpsPort + "/balancer-manager" + "?b=" + balancerName + "&nonce=" + nonce + "&w=" + workerName;
     }
 
     public Map<String, String> findBalancers(final String content) {
         LOGGER.info("Entering findBalancers:");
         Map<String, String> balancers = new HashMap<>();
-        final String foundStringPrefix = "<h3>LoadBalancer Status for <a href=\"/balancer-manager\\?b=";
-        final String foundStringPost = "\\&nonce=";
-        final String foundStringPostNonce = "\">balancer://";
-        final String matchPattern = foundStringPrefix + ".*" + foundStringPost + ".*";
+        final String foundStringPrefix = "<h3>LoadBalancer Status for <a href=..balancer-manager\\?b=";
+        final String foundStringPost = "nonce=";
+        final String foundStringPostNonce = ">balancer";
+        final String matchStringBalancerNamePrefix = "?b=";
+        final String matchPattern = foundStringPrefix + ".*." + foundStringPost + ".*";
         String balancerName;
         String nonce;
         Pattern pattern = Pattern.compile(matchPattern);
         Matcher matcher = pattern.matcher(content);
         String matchString;
         while (matcher.find()) {
             matchString = matcher.group();
-            balancerName = matchString.substring(matchString.indexOf(foundStringPrefix.replace("\\", "")) + foundStringPrefix.replace("\\", "").length(),
-                    matchString.indexOf(foundStringPost.replace("\\", "")));
-            nonce = matchString.substring(
-                    matchString.indexOf(foundStringPost.replace("\\", "")) + foundStringPost.replace("\\", "").length(),
-                    matchString.indexOf(foundStringPostNonce));
+            balancerName = matchString.substring(matchString.indexOf(matchStringBalancerNamePrefix) +
+                    matchStringBalancerNamePrefix.length(), matchString.indexOf("&"));
+            nonce = matchString.substring(matchString.indexOf(foundStringPost) + foundStringPost.length(),
+                    matchString.indexOf(foundStringPostNonce) - 1);
             balancers.put(balancerName, nonce);
         }
+        LOGGER.info("balancers.size: " + balancers.size());
+        for (Map.Entry entry : balancers.entrySet()) {
+            LOGGER.info("balancerName: " + entry.getKey() + ", nonce: " + entry.getValue());
```

Comment:
Can we move this logging statement to line 43 inside the while loop? We shouldn't have logic specifically for logging, especially since we capture all this data in the while loop while building the balancers set.